pkg/mod/mod.go

```diff
@@ -322,66 +315,7 @@ func applyAction(ctx context.Context, us user.Store, cs user.ChatStore,
 		}
 	}
 
-	actionEmailText, ok := ModActionEmailMap[action.Type]
-	if ok {
-		if mailgunKey != "" {
-			emailContent, err := instantiateEmail(user.Username,
-				actionEmailText,
-				action.Note,
-				action.StartTime,
-				action.EndTime,
-				action.EmailType)
-			if err == nil {
-				_, err := emailer.SendSimpleMessage(mailgunKey,
-					user.Email,
-					fmt.Sprintf("Woogles Terms of Service Violation for Account %s", user.Username),
-					emailContent)
-				if err != nil {
-					// Errors should not be fatal, just log them
-					log.Err(err).Str("userID", user.UUID).Msg("mod-action-send-user-email")
-				}
-			} else {
-				log.Err(err).Str("userID", user.UUID).Msg("mod-action-generate-user-email")
-			}
-
-		}
-		if discordToken != "" {
-			modUser, err := us.GetByUUID(ctx, action.ApplierUserId)
-			if err != nil {
-				log.Err(err).Str("userID", user.UUID).Msg("mod-action-applier")
-			} else {
-				message := fmt.Sprintf("Action %s was applied to user %s by moderator %s.", actionEmailText, user.Username, modUser.Username)
-				if !actionExists { // Action is non-transient
-					if action.Duration == 0 {
-						message += " This action is permanent."
-					} else if action.EndTime == nil {
-						log.Err(err).Str("userID", user.UUID).Msg("mod-action-endtime-nil")
-					} else {
-						golangActionEndTime, err := ptypes.Timestamp(action.EndTime)
-						if err != nil {
-							log.Err(err).Str("error", err.Error()).Msg("mod-action-endtime-conversion")
-						} else {
-							message += fmt.Sprintf(" This action will expire on %s.", golangActionEndTime.UTC().Format(time.UnixDate))
-						}
-					}
-				}
-				requestBody, err := json.Marshal(map[string]string{"content": message})
-				// Errors should not be fatal, just log them
-				if err != nil {
-					log.Err(err).Str("error", err.Error()).Msg("mod-action-discord-notification-marshal")
-				} else {
-					resp, err := http.Post(discordToken, "application/json", bytes.NewBuffer(requestBody))
-					// Errors should not be fatal, just log them
-					if err != nil {
-						log.Err(err).Str("error", err.Error()).Msg("mod-action-discord-notification-post-error")
-					} else if resp.StatusCode != 204 { // No Content
-						// We do not expect any other response
-						log.Err(err).Str("status", resp.Status).Msg("mod-action-discord-notification-post-bad-response")
-					}
-				}
-			}
-		}
-	}
+	notify(ctx, us, user, action)
 	return us.Set(ctx, user)
```

Comment:
not sure if `notify` should be before `us.Set` succeeds

Reply:
`notify` after `us.Set` makes more sense. I'll update it.